File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
            map<int,int> mp;
            vector<int> vec;
            int deff;
            for (int i = 0; i < nums.size(); ++i)
            {
                int num = nums[i];
                deff = target - num;
                auto it = mp.find(deff) ;
                if(it != mp.end())
                {
                    vec.push_back(it->second);
                    vec.push_back(i);
                    return vec;
                }
                mp[num] = i;
            }
            return vec;
       
    }
};

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map1 = new Map;
    let array;
    for(let i = 0 ; i <nums.length;i++)
    {
        let num = nums[i];

        let deff = target - num;
        if(map1.has(deff))
        {
            array = [map1.get(deff),i];
            return array;
        }
        map1.set(num,i);
    }
    return []
}